testing/citest/tests/gcs_pubsub_gae_test.py

```diff
@@ -0,0 +1,453 @@
+
+    group_name = frigga.Naming.server_group(
+        app=self.TEST_APP,
+        stack=self.bindings['TEST_STACK'],
+        version='v000')
+
+    # Triggered pipeline does a deploy, check for that server group.
+    server_group_path = 'applications/{app}/serverGroups'.format(app=self.TEST_APP)
+    builder = st.HttpContractBuilder(self.agent)
+    (builder.new_clause_builder('GAE Deploy Pipeline Succeeded',
+                                retryable_for_secs=300)
+     .get_url_path(server_group_path)
+     .EXPECT(
+       ov_factory.value_list_path_contains(
+         None,
+         jp.DICT_SUBSET({'name': group_name})
+       )
+     )
+    )
+
+    executions_path = 'executions?pipelineConfigIds={}&limit=1&statuses=SUCCEEDED'.format(self.__pipeline_id)
+    return st.OperationContract(
+      self.__gcs_pubsub_agent.new_gcs_pubsub_trigger_operation(
+        gate_agent=self.agent,
+        title='monitor_gcs_pubsub_pipeline',
+        bucket_name=self.bucket,
+        upload_path='{name}'.format(name=name),
+        local_filename=os.path.abspath(name),
+        status_class=None,
+        status_path=executions_path
+      ),
+      contract=builder.build())
+
+  def delete_load_balancer(self):
+    bindings = self.bindings
+    payload = self.agent.make_json_payload_from_kwargs(
+        job=[{
+            'type': 'deleteLoadBalancer',
+            'cloudProvider': 'appengine',
+            'loadBalancerName': self.__lb_name,
+            'account': bindings['SPINNAKER_APPENGINE_ACCOUNT'],
+            'credentials': bindings['SPINNAKER_APPENGINE_ACCOUNT'],
+            'user': '[anonymous]'
+        }],
+        description='Delete Load Balancer: {0} in {1}'.format(
+            self.__lb_name,
+            bindings['SPINNAKER_APPENGINE_ACCOUNT']),
+        application=self.TEST_APP)
+
+    builder = gcp.GcpContractBuilder(self.appengine_observer)
+    (builder.new_clause_builder('Service Deleted', retryable_for_secs=30)
+     .inspect_resource('apps.services',
+                       self.__lb_name,
+                       appsId=self.__gcp_project)
+     .EXPECT(
+         ov_factory.error_list_contains(gcp.HttpErrorPredicate(http_code=404))))
+
+    return st.OperationContract(
+        self.new_post_operation(
+            title='delete_load_balancer', data=payload, path='tasks'),
+        contract=builder.build())
+
+  def delete_deploy_pipeline(self, pipeline_id):
+    payload = self.agent.make_json_payload_from_kwargs(id=pipeline_id)
+    path = os.path.join('pipelines', self.TEST_APP, pipeline_id)
+
+    builder = st.HttpContractBuilder(self.agent)
+    (builder.new_clause_builder('Has Pipeline',
+                                retryable_for_secs=5)
+     .get_url_path(
+       'applications/{app}/pipelineConfigs'.format(app=self.TEST_APP))
+     .excludes_path_value('name', pipeline_id))
+
+    return st.OperationContract(
+      self.new_delete_operation(
+        title='delete_bake_pipeline', data=payload, path=path,
```

Comment:
Is this really synchronous?
Also you call it "bake_pipeline" but it is a deploy pipeline.

Reply:
Yes AFAIK.

Reply:
Fixed naming.